scripts/perftest/python/utils.py

```diff
@@ -301,4 +304,86 @@ def get_folder_metrics(folder_name, action_mode):
     except IndexError:
         intercept = 'none'
 
-    return mat_type, mat_shape, intercept
\ No newline at end of file
+    return mat_type, mat_shape, intercept
+
+
+def mat_type_check(current_family, matrix_type):
+    """
+    Some Algorithms support different matrix_type. This function give us the right matrix_type given
+    an algorithm
+
+    current_family: String
+    Current family being porcessed in this function
+
+    matrix_type: List
+    Type of matrix to generate dense, sparse, all
+
+    return: List
+    Return the list of right matrix types supported by the family
+    """
+    family_no_matrix_type = ['clustering', 'stats1', 'stats2']
```

Comment:
Hardcoding which families do not need sparse datagen in a util function does not seem like a good idea to me.
Place it with the other config (in `run_perftest.py` where `ML_ALGO` and other configuration is specified).